scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHello.java

```diff
@@ -98,46 +102,48 @@
 	/**
 	 * Creates a <em>Client Hello</em> message to be sent to a server.
 	 * 
-	 * @param version
-	 *            the protocol version to use
-	 * @param secureRandom
-	 *            a function to use for creating random values included in the
-	 *            message
-	 * @param supportedClientCertificateTypes
-	 *            the list of certificate types supported by the client
-	 * @param supportedServerCertificateTypes
-	 *            the list of certificate types supported by the server
-	 * @param peerAddress
-	 *            the IP address and port of the peer this message has been
-	 *            received from or should be sent to
+	 * @param version the protocol version to use
+	 * @param secureRandom a function to use for creating random values included
+	 *            in the message
+	 * @param cipherSuites the list of the supported cipher suites
```

Comment:
please rename to `preferredCipherSuites` to better reflect the purpose of the cipher suites and adapt the description accordingly

Reply:
FMPOV `preferredCipherSuites` is the WRONG term! According RFC5246, 7.4.1.2, CLIENT_HELLO, the cipher suites are just ORDERED according the preference. 

>The cipher suite list, passed from the client to the server in the
> ClientHello message, contains the combinations of cryptographic
> algorithms supported by the client in order of the client’s
> preference (favorite choice first).
> ...
> The server will select a cipher
> suite or, if no acceptable choices are presented, return a handshake
> failure alert and close the connection.

So the client do not prefer the cipher suites, it only supports those in the list.